aten/src/ATen/native/quantized/cpu/qnnpack_utils.h

```diff
@@ -2,11 +2,62 @@
 
 #ifdef USE_PYTORCH_QNNPACK
 #include <pytorch_qnnpack.h>
+#include <qnnpack_func.h>
 
 struct QnnpackOperatorDeleter {
   void operator()(pytorch_qnnp_operator_t op) {
     pytorch_qnnp_delete_operator(op);
   }
 };
+struct PackedLinearWeightsQnnp {
+  std::unique_ptr<qnnpack::PackBMatrix> w;
+  double w_scale;
+  int64_t w_zp;
+};
+
+enum class Activation : uint8_t { NONE = 0, RELU = 1 };
+
+#if defined(__ANDROID__) && !defined(__NDK_MAJOR__)
+template <class T>
+inline float Round(const float x) {
+  return ::nearbyintf(x);
+}
+inline double Round(const double x) {
+  return ::nearbyint(x);
+}
+#else
+template <class T>
+inline T Round(const T x) {
+  return std::nearbyint(x);
+}
+#endif
 
+inline uint8_t QuantizeUint8(float scale, int32_t zero_point, float value) {
```

Comment:
btw, this seems to be similar to fbgemm implementation used in quantize_val. Can they be the same?

Reply:
Looks similar, but that function calls `fbgemm::Quantize` function. I wanted to avoid calling fbgemm functions from pytorch mobile build because I don't think they will be built on mobile.